### PR TITLE
copr: vectorize sub_date_and_string, sub/add date and duration

### PR DESCRIPTION
**Pull request by iliveforfun**
Signed-off-by: zhuhaozhuhao notmyfault1996@gmail.com

### What problem does this PR solve?

Issue Number: #9016 

Problem Summary: This PR is for vectorizing functions in impl_time.rs

### What is changed and how it works?

What's Changed:
what is changed: add vectorize sub_date_and_string, sub/add date and duration and related test.

### Related changes

None

### Check List 

Tests 

- Unit test

Side effects

### Release note 
- Vectorize sub_date_and_string, sub/add date and duration in tikv coprocessor functions

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi, please rebase your branch and don't include commits from already merged PRs.

**Comment by iliveforfun** [comment]
> Hi, please rebase your branch and don't include commits from already merged PRs.

Sry, for the reason that I keep the local branch last time, so I have to use `git push -f` to replace the commit history.

**Comment by Xuanwo** [comment]
Yes, this time we do need a `force push`.

**Review by Xuanwo — APPROVED** [review]

File: components/tidb_query_expr/src/impl_time.rs [hunk 1]
```diff
@@ -443,6 +465,55 @@ pub fn sub_datetime_and_string(
     Ok(Some(res))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn sub_date_and_string(
```
Comment (skyzh) [inline components/tidb_query_expr/src/impl_time.rs:470]: Seems that `sub_date_and_string` should return a string. Would you please double-check whether I'm correct? Refer to https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L6590 for more information.

Comment (skyzh) [reply components/tidb_query_expr/src/impl_time.rs:470]: Also https://github.com/tikv/tikv/blob/1a88f12ebf50064b992fc5efc7e7f55795210521/components/tidb_query_normal_expr/src/builtin_time.rs#L595

**Review by skyzh — COMMENTED** [review]
Seems that we have implemented a lot of the datetime related functions with the wrong return type before 🤣🤣🤣, and I'm afraid I could not let any other maybe-incorrect implementation be merged. I'll investigate this later, and you could also help us find out what return type should those functions be and fix them :) Thanks!

**Comment by skyzh** [comment]
And that's weird that integration-copr-test doesn't cover these functions last time https://github.com/tikv/tikv/pull/9921 . Maybe we should also add them to push-down list and do some integration tests? https://github.com/tikv/copr-test/tree/master/push-down-test

File: components/tidb_query_expr/src/impl_time.rs (hunk 1 above)
Comment (iliveforfun) [reply components/tidb_query_expr/src/impl_time.rs:470]: ok, I will check them.

**Comment by Xuanwo** [comment]
Thank you for your effort thus far. However, this PR has been inactive for two years without any progress. Therefore, I have decided to close it in order to free up space. You are welcome to reopen the PR if you wish to continue working on it.

Thanks again!

**Pull request closed without merging**